tests/system/src/test/java/test/javafx/scene/text/ArabicWrappingTest.java

```diff
@@ -0,0 +1,214 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.text;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+import javafx.application.Platform;
+import javafx.stage.Stage;
+import javafx.scene.Scene;
+import javafx.scene.layout.HBox;
+import javafx.scene.text.Font;
+import javafx.scene.text.Text;
+import javafx.application.Application;
+
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class ArabicWrappingTest extends Application {
+
+    static PrintStream systemErr = System.err;
+    static SystemErrFilter systemErrFilter;
+
+    static class SystemErrFilter extends PrintStream {
+        private boolean foundException = false;
+        private String exceptionMsg;
+
+        public SystemErrFilter(OutputStream out) {
+            super(out);
+        }
+
+        @Override
+        public synchronized void print(String s) {
+            System.out.flush();
+            if (s.indexOf("Exception") >= 0) {
+                foundException = true;
+                exceptionMsg = s;
+            }
+            super.print(s);
+        }
+
+        boolean checkException() {
+            return foundException;
+        }
+
+        String getExceptionString() {
+            return exceptionMsg;
+        }
+    }
+
+    @BeforeClass
+    public static void initFX() {
+        systemErrFilter = new SystemErrFilter(System.err);
+        System.setErr(systemErrFilter);
+        new Thread(() -> {
+            Application.launch(ArabicWrappingTest.class);
+        }).start();
+    }
+
+    @AfterClass
+    public static void exitTest() {
+        Platform.exit();
+    }
+
+    static volatile boolean testDone = false;
+    static volatile boolean testPassed;
+
+    @Test
```

Comment:
Can you add a timeout value? That way if something goes wrong the test won't hang indefinitely. Something like:

```
    @Test (timeout=60000)
```

Reply:
Will do. I actually had one in there .. but I was having some problems with @Test not being recognised and I removed it in case it was part of the problem. I will put it back.